docs/docs/mdx/writing-pages.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Writing Pages in MDX
+---
+
+After [installing](/docs/mdx/getting-started) the plugin, MDX files
+written in `src/pages` will turn into pages. This happens because
+`gatsby-mdx` looks for MDX files and automatically transpiles them
+so that Gatsby internals can render them.
+
+Pages are rendered at a URL that is constructed from the filesystem
+path inside `src/pages`. An MDX file at `src/pages/awesome.mdx` will
+result in a page being rendered at `mysite.com/awesome`.
+
+By default, gatsby-mdx supports frontmatter so you can define things
+like titles and paths to use in your GraphQL queries.
+
```

Comment:
Hey, 

Complete newbie here, but I think this is a bit confusing, because in lines 5-12, you explaining about the auto-creation of pages in the `src/pages` directory, and then you go right explaining that by default gatsby-mdx will parse the frontmatter, and it might easily appear to the uninitiated that you can use frontmatter in MDX files in `src/pages` directory, and I don't believe that is the case---at least, I have not been able to get it to work.

Reply:
you can use frontmatter in MDX files with `gatsby-mdx`. We added that feature on top of MDX to make it easier for people to convert from Remark. File an issue at https://github.com/ChristopherBiscardi/gatsby-mdx/tree/master/packages/gatsby-mdx with a minimal repro for help in getting it to work.

Reply:
Hey @ChristopherBiscardi, Thanks for the kind reply. I probably should have found a better place to comment than a code review on a PR, but I wasn't sure where! 😀 I get what you are saying, and I have that working well. It's great. The problem I see is that the first 3 paragraphs of this file will not be clear to a newbie to gatsby such as myself. As I understand it, the files in `pages/src` are automatically added to the site, but you cannot query them with graphQL, because they are not added to `allMdx`. This makes perfect sense as a design decision.

Frontmatter is automatically parsed and available via graphQL on files processed  via `gatsby-source-filesystem`. That also makes sense. I just think that there could be a bit more clarity in these three paragraphs. I know once one grasps the gatsby way of thinking, this is obvious, but until then it might be a bit confusing.